app/src/main/java/com/cliqz/browser/settings/GeneralSettingsFragment.kt

```diff
@@ -0,0 +1,315 @@
+package com.cliqz.browser.settings
+
+import acr.browser.lightning.constant.SearchEngines
+import android.app.Activity
+import android.content.Intent
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import android.preference.*
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import com.cliqz.browser.BuildConfig
+import com.cliqz.browser.R
+import com.cliqz.browser.main.Countries
+import com.cliqz.browser.main.MainActivity
+import com.cliqz.browser.telemetry.TelemetryKeys
+
+class GeneralSettingsFragment : BaseSettingsFragment() {
+
+    private var startTime: Long = 0
+
+    private var mActivity: Activity? = null
+    private var searchEngine: Preference? = null
```

Comment:
We can avoid declaring properties as '_Nullable_' (with ?) by marking them as `lateinit`.
This helps to avoid null checks on the properties (as done on line 63)
So this becomes
```
private lateinit var searchEngine: Preference
```
And line 63 becomes:
```
searchEngine.onPreferenceClickListener = this
```
When you are absolutely sure that the property will be initialized before it is accessed, marking it as `lateinit` is a good idea.

[Nullable types and Non-Null Type](https://kotlinlang.org/docs/reference/null-safety.html#nullable-types-and-non-null-types)
[Late-Initialized Properties and Variables](https://kotlinlang.org/docs/reference/properties.html#late-initialized-properties-and-variables)

[Kotlin - How to decide between “lateinit” and “nullable variable”?](https://stackoverflow.com/a/44796205/2686502)

Reply:
Not is this case: because the xmls are different between Lumen and Cliqz, I do not know what will result in null reference and what not. Obviously I can decide case by case but, for the sake of uniformity, I decided to not do so.